Firmware/Marlin_main.cpp

```diff
@@ -7638,16 +7640,72 @@ SERIAL_PROTOCOLPGM("\n\n");
 
     /*!
     ### M601 - Pause print <a href="https://reprap.org/wiki/G-code#M601:_Pause_print">M601: Pause print</a>
+    Without any parameters it will park the extruder to default.
+    #### Usage
+
+         M601 [ X | Y | Z | S ]
+
+    #### Parameters
+     - `X` - X position to park at (otherwise X_PAUSE_POS 50) these are saved until change or reset.
+     - `Y` - Y position to park at (otherwise Y_PAUSE_POS 190) these are saved until change or reset.
+     - `Z` - Z raise before park (otherwise Z_PAUSE_LIFT 20) these are saved until change or reset.
+     - `S` - Set values [S0 = set default | S1 = set values only without pausing]
     */
     /*!
-    ### M125 - Pause print (TODO: not implemented)
+
+    ### M125 - Pause print <a href="https://reprap.org/wiki/G-code#M125:_Pause_print">M125: Pause print</a>
+    Without any parameters it will park the extruderthe extruder to default.
+    #### Usage
+
+         M125 [ X | Y | Z | S ]
+
+    #### Parameters
+     - `X` - X position to park at (otherwise X_PAUSE_POS 50) these are saved until change or reset.
+     - `Y` - Y position to park at (otherwise Y_PAUSE_POS 190 these are saved until change or reset.
+     - `Z` - Z raise before park (otherwise Z_PAUSE_LIFT 20) these are saved until change or reset.
+     - `S` - Set values [S0 = set to default values | S1 = set values only without pausing]
     */
     /*!
     ### M25 - Pause SD print <a href="https://reprap.org/wiki/G-code#M25:_Pause_SD_print">M25: Pause SD print</a>
     */
     case 25:
+    case 125:
     case 601:
     {
+        if (code_seen('X')) {
+          //Check that the X pause position is within printer limits
+          if ((code_value() >= X_MIN_POS) && (code_value() <= X_MAX_POS-1)) {
```

Comment:
Why -1?

Reply:
In MK404 the `M125 X255` causes a crash and I just wanted to eliminate that may happen on real hardware as well.
Selftest on my MK3S shows
```
Recv: Measured axis length:255.570
Recv: Measured axis length:255.570
Recv: Axis length difference:0.000

Recv: Measured axis length:220.370
Recv: Measured axis length:220.690
Recv: Axis length difference:0.320
```

So my printer would not crash with `M125 X255` but others may